### PR TITLE
Fix splitting

### PR DESCRIPTION
**Pull request by ggeier**

### CODE REVIEW

File: src/tools/tslsplit/Main.hs [hunk 1]
```diff
@@ -19,36 +19,73 @@ module Main
...
 
 -----------------------------------------------------------------------------
 
+-- | The data type contains all flags and settings
+-- that can be adjusted to influence the behavior of the library:
+data Configuration =
+  Configuration
+  { -- | The input file containing the synthesized control flow
+    -- model.
+    inputFile :: FilePath
```
Comment (stemar94) [inline src/tools/tslsplit/Main.hs]: In the other tsltools, the input is an optional argument. See e.g. tslsym sources.

File: src/tools/tslsplit/Main.hs [hunk 2]
```diff
@@ -19,36 +19,73 @@ module Main
...
+  , -- | A Boolean flag specifying whether to split assumptions
+    -- for the assumption violation check
+    -- or perform usual specification decomposition.
+    assumptions :: Bool
+  } deriving (Eq, Ord)
+
+configParser :: Parser Configuration
+configParser = Configuration
+  <$> argument str
```
Comment (stemar94) [inline src/tools/tslsplit/Main.hs]: see above

File: src/lib/TSL/Splitter.hs [hunk 3]
```diff
@@ -126,7 +126,7 @@ split spec@Specification{assumptions, guarantees} =
     splitGuars = splitFormulas guarantees connComp
     splitAssumptions  = splitFormulas boundAssumptions connComp
   in
-    fmap cleanSymboltable $ buildSpecs $ addFreeAssumptions freeAssumptions $ zip splitAssumptions splitGuars
+    fmap cleanSymboltable $ buildSpecs $ addFreeAssumptions freeAssumptions $ filter (not . null . snd) $ zip (splitAssumptions ++ repeat []) splitGuars
```
Comment (stemar94) [inline src/lib/TSL/Splitter.hs]: How comes that |splitAssumptions| /= |splitGuars| is possible?

File: src/lib/TSL/Splitter.hs [hunk 4]
```diff
@@ -124,13 +125,32 @@ split spec@Specification{assumptions, guarantees} =
     connComp = map Set.fromList $ components graph
     (freeAssumptions, boundAssumptions) = partition (null . (intersection decRelProps) . getInOutputs) assumptions
     splitGuars = splitFormulas guarantees connComp
```
Comment (stemar94) [inline src/lib/TSL/Splitter.hs]: I would patch this variable naming as well, if you choose to fix for assumptions.

File: src/lib/TSL/Splitter.hs (hunk 3 above)
Comment (ggeier) [reply src/lib/TSL/Splitter.hs]: Both lists have as many elements as connected components in the dependency graph plus all formulas that do not match any component. These could be e.g. `true` or `false` as they do not depend on any variable. For guarantees we may not ignore a `false`. For assumption, however we can ignore a `false` as we assume unrealizability of `assumption -> false`.

File: src/lib/TSL/Splitter.hs (hunk 4 above)
Comment (ggeier) [reply src/lib/TSL/Splitter.hs]: done in ba2a725

File: src/tools/tslsplit/Main.hs (hunk 1 above)
Comment (ggeier) [reply src/tools/tslsplit/Main.hs]: done in 642955b

File: src/tools/tslsplit/Main.hs (hunk 2 above)
Comment (ggeier) [reply src/tools/tslsplit/Main.hs]: done in 642955b

File: src/tools/tslsplit/Config.hs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import Options.Applicative
+
+-----------------------------------------------------------------------------
+
+-- | The data type contains all flags and settings
+-- that can be adjusted to influence the behavior of the library:
+data Configuration =
+  Configuration
+  { -- | The input file containing the specification
+    inputFile :: Maybe FilePath
```
Comment (stemar94) [inline src/tools/tslsplit/Config.hs]: just 'input', again for consistency reasons only

Comment (ggeier) [reply src/tools/tslsplit/Config.hs]: done 1426db2

**Comment by stemar94** [comment]
LGTM

**Comment by stemar94** [comment]
Could you provide, a small commit description, so that I could squash these 7 commits into 1.

**Pull request merged**